### PR TITLE
fix muxAnim function

### PR DESCRIPTION
**Pull request by mpirescarvalho**
This PR fixes problems with setting exif data to animated webp files. It's basically some missing/wrong params.

### CODE REVIEW

File: webp.mjs [hunk 1]
```diff
@@ -297,7 +297,7 @@ class Image {
           break;
       }
       cursor += header.size+1;
-      if (header.size&1) { cursor++; }
+      // if (header.size&1) { cursor++; }
```
Comment (mpirescarvalho) [inline webp.mjs:300]: I'll explaing this change so you can decide what to do with it

I was getting this error:
```bash
(node:3454) UnhandledPromiseRejectionWarning: Error: Unknown image type
    at Function.muxAnim (file:///home/marcelo/www/test/node_modules/node-webpmux/webp.mjs:541:24)
    at Image.muxAnim (file:///home/marcelo/www/test/node_modules/node-webpmux/webp.mjs:178:126)
    at /home/marcelo/www/test/index.js:8:13
```

From my investigation, this switch was never matching any case, therefore the frames didn't had the type filled: 
https://github.com/ApeironTsuka/node-webpmux/blob/9d01fb7f5fd514253c404a4a7c07f800e13b8ad8/webp.mjs#L275

I log the values of `header` and got this values:

```bash
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 >', size: 1342177318 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8  ', size: 268435494 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 ,', size: 4026531878 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 1879048231 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 T', size: 2952790057 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 r', size: 1879048233 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 1342177321 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 2415919145 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 T', size: 3489660969 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 3489660969 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 |', size: 805306409 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 3489660969 }
{ fourCC: 'ALPH', size: 57 }
{ fourCC: 'P8 �', size: 4026531882 }
```

So the `fourCC` was never matching the swtich case values, therefore the frame ended up like this:
```js
{
  raw: <Buffer 00 00 00 00 00 00 ff 01 00 ff 01 00 64 00 00 00 41 4c 50 48 39 00 00 00 01 17 20 10 48 c2 dd 5f 60 8d 88 08 07 45 6d a4 40 98 04 67 88 be fe 8d e8 ff ... 10972 more bytes>,
  x: 0,
  y: 0,
  width: 512,
  height: 512,
  duration: 100,
  blend: true,
  dispose: false
}
```

**Comment by ApeironTsuka** [comment]
The reason for that line is that if a chunk's size is odd, there's an extra padding byte at the end that must be skipped over. Or at least that's how it's supposed to work. I might've misunderstood and the rule doesn't apply in nested chunks (such as in the case of animations). I'll add this change with a couple others.

**Comment by mpirescarvalho** [comment]
It doesn't worked very well, the image lost its width and height. 
I think passing `this.width` and `this.height` is necessary:
image

Also, I don't think setting exif data worked, but I'm out of time and haven't tested it enough so it could be something on my end. I'll run some tests tomorow and post here the results

**Comment by mpirescarvalho** [comment]
By the way, reading ANMF chunk worked

**Comment by ApeironTsuka** [comment]
Should be fixed properly now. I hadn't noticed line 299 (`cursor += header.size+1;`) had that +1 before that should _not_ have been there. Reading ANMF chunks should work properly now. Writing ANMF chunks should also now work correctly.

As for .muxAnim not having width/height set, that's because frames could have been added/removed/changed that would then change the animation's final dimensions. Not passing width/height causes Image.muxAnim to recalculate them. I added width and height overrides to .muxAnim in case it's needed/desired.

For saving EXIF, remember that you must tell .muxAnim that you want it saved. `img.muxAnim({ path: 'output.webp', exif: true, loops: img.anim.loopCount })` for instance. This passes img's EXIF to Image.muxAnim.

**Comment by mpirescarvalho** [comment]
I'm sorry, I've got it wrong! 

In fact my width and height are both identical for the input and output image. It turns out it's the background that is wrong

Take a look at this:

The source image has a transparent background
image

But it is outputing a black background
image

I've tried playing arround with param `bgColor` in `muxAnim()` but it didn't make any difference

**By the way, the exif data it's working perfectly**

**Comment by ApeironTsuka** [comment]
Pushed what should be a fix. It wasn't reading ALPH chunks at all as for some reason I only did a codepath for if the ALPH chunk came after VP8, even though the "correct" case is ALPH coming before it.

**Comment by mpirescarvalho** [comment]
It didn't work, still losing transparency 😕

I'm using it like this:
```js
  const img = new NodeWebP.Image();
  await img.load('./video.webp');
  const exif = fs.readFileSync('./exif.exif');
  img.exif = exif
  await img.muxAnim({ path: './sticker.webp', exif: true })
```

Result is the same as my previous comment

**Comment by ApeironTsuka** [comment]
Pushed an update changing how .muxAnim works a little bit. It will now use the image's already-defined background color, loop count, width/height, and automatically pass EXIF/ICCP/XMP when available. This is how it should've been working all along, honestly. No longer need to pass exif: true, and instead would need to pass exif: false if you _don't_ want it preserved.

I think what went on this time is it was using the default (should be white) background in the output, maybe? I did some testing with animations that were using ALPH+VP8 (lossy with alpha) and VP8L (lossless with alpha) frames on transparent backgrounds, and both came out as expected this time.

I haven't been pushing these last handful of changes to NPM. Now they're up as 1.1.5, just in case.

**Comment by mpirescarvalho** [comment]
> I haven't been pushing these last handful of changes to NPM. Now they're up as 1.1.5, just in case.
I know, I'm pulling from this repo

It is still outputing a black background. I can push the source and the output to a repo if you want to take a look at the files

**Comment by ApeironTsuka** [comment]
That would help, if you can. I'm not sure what's going on now.

**Comment by mpirescarvalho** [comment]
Just uploaded: `input.webp` and `output.webp`

https://github.com/mpirescarvalho/node-webpmux-test

**Comment by ApeironTsuka** [comment]
Definitely fixed now, thanks. Hopefully you don't run into any more issues. I definitely need to put together a large suite of test images to throw at this later, as my existing testing has clearly not been enough.

**Comment by mpirescarvalho** [comment]
Working now, thanks!

**Comment by ApeironTsuka** [comment]
Was waiting in case something else cropped up. Pushing now.

**Pull request closed without merging**